src/index.ts

```diff
@@ -279,21 +279,41 @@ class Server {
           data.context = ('vessels.' + app.selfId) as Context
         }
         const now = new Date()
-        data.updates.forEach((update: Update) => {
-          if (typeof update.source !== 'undefined') {
-            update.source.label = providerId
-            if (!update.$source) {
-              update.$source = getSourceId(update.source)
+        data.updates = data.updates
+          .map((update: Partial<Update>) => {
+            if (typeof update.source !== 'undefined') {
+              update.source.label = providerId
+              if (!update.$source) {
+                update.$source = getSourceId(update.source)
+              }
+            } else {
+              if (typeof update.$source === 'undefined') {
+                update.$source = providerId as SourceRef
+              }
             }
-          } else {
-            if (typeof update.$source === 'undefined') {
-              update.$source = providerId as SourceRef
+            if (!update.timestamp || app.config.overrideTimestampWithNow) {
+              update.timestamp = now.toISOString() as Timestamp
             }
-          }
-          if (!update.timestamp || app.config.overrideTimestampWithNow) {
-            update.timestamp = now.toISOString() as Timestamp
-          }
-        })
+
+            if ('values' in update && !Array.isArray(update.values)) {
+              debug(`handleMessage: ignoring invalid values`, update.values)
+              delete update.values
+            }
+
+            if ('meta' in update && !Array.isArray(update.meta)) {
+              debug(`handleMessage: ignoring invalid meta`, update.meta)
+              delete update.meta
+            }
+
+            if ('values' in update || 'meta' in update) {
+              return update as Update
+            }
+          })
+          .filter((update) => update !== undefined)
+
+        // No valid updates, discarding
+        if (data.updates.length < 1) return
+
```

Comment:
Prettier's 80 char line limit is so annoying. Most of these changes are because line 282 is now just over 80 characters, and now it's hard to see what else I actually changed. Anyone opposed to changing it to 100 characters?

Here's a link to [view the diff without whitespace changes](https://github.com/SignalK/signalk-server/pull/1945/files?w=1)